samples/src/Cache/CacheAspect.cs

```diff
@@ -50,16 +54,17 @@ public class CacheAspect
             object result = null;
             var resultFound = false;
 
-            var cacheTriggers = triggers.OfType<CacheAttribute>();
-            var key = GetKey(instance, target.Method, args);
+            var cacheTriggers = (triggers.OfType<CacheAttribute>()).ToList();
+            var useInstance = cacheTriggers.Any(ct => ct.PerInstanceCache);

```

Comment:
This then should be calculated on per cacheTrigger inside foreach block.
Reason is - the aspect is basically a processor that executes caching logic for all types of cache. And there could be several types of cache simultaneously, like ```MemoryCache(10, PerIntsanceCache= true)``` and ```MemoryCache(300, PerIntsanceCache= false)``` and probably ```RedisCache(9000, PerIntsanceCache= true) for the same method.

Reply:
I've modified it to be calculated per trigger. It's a bit inefficient that I calculate both keys upfront, but this way it stays readable and easy to override.

Reply:
I agree, slightly inefficient, but what are our options?